src/test/java/com/redhat/devtools/lsp4ij/features/formatting/GoClientSideFormatOnCloseBraceTest.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.redhat.devtools.lsp4ij.client.features.LSPFormattingFeature.FormattingScope;
+import com.redhat.devtools.lsp4ij.fixtures.LSPClientSideOnTypeFormattingFixtureTestCase;
+
+/**
+ * Go-based client-side on-type formatting tests for format-on-close brace. The Go language server does not support
+ * range formatting, so these tests confirm its behavior when configured for file scope and also that it does nothing
+ * when configured for code block scope (i.e., the default scope).
+ */
+public class GoClientSideFormatOnCloseBraceTest extends LSPClientSideOnTypeFormattingFixtureTestCase {
```

Comment:
Added this Go language server-specific test to verify behavior for a language server that doesn't support range formatting but does support file formatting.